File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.py
class Solution:
    def validStrings(self, n: int) -> List[str]:
        def backtrack(i, sl):
            if i == n-1:
                res.append("".join(sl))
                return

            if sl[i] == '0':
                backtrack(i+1, sl+['1'])
            else:
                backtrack(i+1, sl+['1'])
                backtrack(i+1, sl+['0'])

        res = []
        backtrack(0, ['1'])
        backtrack(0, ['0'])
        return res

File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.cpp
class Solution {
public:
    vector<string> res;
    void backtrack(int n, int i, string& subs) {
        if (i == n-1) {
            res.push_back(subs);
            return;
        }
        subs.push_back('1');
        backtrack(n, i+1, subs);
        subs.pop_back();
        if (subs[i] == '1') {
            subs.push_back('0');
            backtrack(n, i+1, subs);
            subs.pop_back();
        }
    }

    vector<string> validStrings(int n) {
        string a="1", b="0";
        backtrack(n, 0, a); 
        backtrack(n, 0, b);
        return res;
    }
};